napari/_vispy/canvas.py

```diff
@@ -332,8 +332,12 @@ def _map_canvas2world(
             of the viewer.
         """
         nd = self.viewer.dims.ndisplay
-        transform = self.view.scene.transform
-        mapped_position = transform.imap(list(position))[:nd]
+        transform = self.view.camera._scene_transform
```

Comment:
what's the difference between `view.scene.transform` and `camera._scene_transform`? Shouldn't they be the same?

Reply:
I don't exactly know, here's my guess. view.scene.transform maps the coordinates from visual(data) to scene, camera._scene_transform transforms scene coordinates to camera coordinates and then project them to canvas.

Reply:
I don't think that's right, I just tried to revert to `self.view.scene.transform` and everything seems to be working fine, also in perspective mode.

Was there a specific reason why you made this change? Unless there's a reason I'm missing, I think it's better to go back and not use private attributes:

```suggestion
        transform = self.view.scene.transform
```